src/transformers/generation_utils.py

```diff
@@ -743,6 +759,47 @@ def _merge_criteria_processor_list(
         default_list.extend(custom_list)
         return default_list
 
+    def compute_transition_beam_scores(
+        self,
+        sequences: torch.Tensor,
+        scores: Tuple[torch.Tensor],
+        beam_indices: torch.Tensor,
+        eos_token_id: int = None,
+        dummy=None,
+    ):
+        """compute the transition probabilities of sequences given generation
+        scores and beam indices"""
+        # reshape scores
+        scores = torch.stack(scores).reshape(len(scores), -1).transpose(0, 1)
```

Comment:
I really like comments explaining the expected shapes of tensors in here to make the code more readable/understandable.

`#reshape scores` is not very informative (since I can read it in the following line, along with `stack` and `transpose`.
What I am missing as a reader is:
- Why you are doing this?
- What should the final result be ?

`# Reshape scores as [vocab_size * batch_size, seq_len] in order to use a single gather op later` is a suggestions